### PR TITLE
Move to dash

### PR DESCRIPTION
**Pull request by pechouc**
This PR aims at moving from the Streamlit to the Dash MVP. 

The Dash MVP is not finalised yet. The following remains to be done:

- Connection to the back-end
- Improving the response of the layout to the type of device being used
- Adding some design elements
- Re-organising the code in separate modules to improve readability (?)

Link to the deployed app: https://pyro-crowdsourcer.herokuapp.com

<details>
<summary><b>How it looks after deployment on desktop (Safari)</b></summary>

Capture d’écran 2022-06-10 à 10 36 46
Capture d’écran 2022-06-10 à 10 36 55
Capture d’écran 2022-06-10 à 10 36 57

</details>

<details>
<summary><b>How it looks after deployment on mobile (Safari)</b></summary>

IMG_4119
IMG_4120
IMG_4121

</details>

### CODE REVIEW

**Comment by pechouc** [comment]
@frgfm, when reviewing, do you think you could look into the Docker set-up now that we have a Dash app, instead of a Streamlit one? I tried to adapt what was done for the `pyro-platform`, but I did not manage to make it work. I went for another solution found online, which might not allow to split the code in different Python files... Many thanks in advance!

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -14,6 +14,6 @@ jobs:
       - name: Build docker image
         run: docker build . -t pyro-crowdsourcer:py3.8.1-slim
       - name: Run docker container
-        run: docker run -d -p 8001:8501 pyro-crowdsourcer:py3.8.1-slim
+        run: docker run -d -p 8001:80 pyro-crowdsourcer:py3.8.1-slim
```
Comment (frgfm) [inline .github/workflows/docker.yml:17]: Are you positive? The last line of "app.py" seems to run the app on port 8050 :thinking:

File: Dockerfile [hunk 2]
```diff
@@ -16,4 +16,6 @@ RUN apt-get update \
 
 COPY ./src src/
 
-CMD ["streamlit", "run", "src/app.py"]
+# CMD ["python", "src/app.py"]
```
Comment (frgfm) [inline Dockerfile:19]: we can remove the comment version afterwards

File: requirements.txt [hunk 3]
```diff
@@ -1 +1,6 @@
-streamlit>=1.1.0
+pandas>=1.3.5
```
Comment (frgfm) [inline requirements.txt:1]: let's remove pandas

File: src/app.py [hunk 4]
```diff
@@ -1,90 +1,886 @@
-# Copyright (C) 2022, Pyronear.
+from datetime import date, datetime
 
-# This program is licensed under the Apache License version 2.
-# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+import dash
+import dash_bootstrap_components as dbc
+import dash_daq as daq
+import pandas as pd
```
Comment (frgfm) [inline src/app.py:6]: replace with
```python
import csv
import requests
```

File: src/app.py [hunk 5]
```diff
@@ -1,90 +1,886 @@
...
-import config
+department_options = []
+departments['to_be_displayed'] = departments['code_departement'] + ' - ' + departments['nom_departement']
+for _, row in departments.iterrows():
+    department_options.append(
+        {
+            'label': row['to_be_displayed'],
+            'value': row['code_departement']
+        }
+    )
```
Comment (frgfm) [inline src/app.py:29]: Can be replaced with

```python
CSV_URL = 'https://static.data.gouv.fr/resources/departements-de-france/20200425-135513/departements-france.csv'


with requests.Session() as s:
    download = s.get(CSV_URL)

    decoded_content = download.content.decode('utf-8')

    cr = csv.reader(decoded_content.splitlines(), delimiter=',', quotechar='"', quoting=csv.QUOTE_ALL)
    data = [{"label": f"{code} - {name}", "value": name} for code, name, _, _ in cr][1:]
```

File: requirements.txt (hunk 3 above)
Comment (frgfm) [reply requirements.txt:1]: and add "requests>=2.20.0"

File: src/app.py [hunk 6]
```diff
@@ -1,90 +1,886 @@
...
+                # html.A(
+                #     dbc.Row(
+                #         [
+                #             dbc.Col(html.Div('Hello')),
+                #         ],
+                #         align="center",
+                #         no_gutters=True,
+                #     ),
+                #     href="#",
+                # ),
```
Comment (frgfm) [inline src/app.py:180]: That can be removed right?

File: src/app.py [hunk 7]
```diff
@@ -1,90 +1,886 @@
...
+                dbc.NavbarToggler(id="navbar-toggler"),
```
Comment (frgfm) [inline src/app.py:181]: we don't need navigation?

File: src/app.py [hunk 8]
```diff
@@ -1,90 +1,886 @@
...
+
+    except ValueError:
+        return False
+
+
+def build_image_upload():
+
+    upload = dcc.Upload(
+        id='image-upload',
+        children=html.Div(
```
Comment (frgfm) [inline src/app.py:51]: As suggested here https://community.plotly.com/t/uploading-relatively-small-files-is-taking-too-long-in-dash-plotly/32785,
let's temporarily save the image to disk before uploading :+1:

**Review by frgfm — CHANGES_REQUESTED** [review]
Thanks a lot!
I added a few comments, we can remove pandas and use "csv" + "requests" to parse it faster and with lighter dependencies :+1:

File: src/app.py (hunk 6 above)
Comment (pechouc) [reply src/app.py:182]: Yes, done.

File: src/app.py (hunk 7 above)
Comment (pechouc) [reply src/app.py:183]: No, and this removes the useless button that we noticed during the call if I am not mistaken.

File: src/app.py (hunk 8 above)
Comment (pechouc) [reply src/app.py:53]: Edit done.

File: src/app.py (hunk 5 above)
Comment (pechouc) [reply src/app.py:31]: Thanks for the snippet!

File: Dockerfile (hunk 2 above)
Comment (pechouc) [reply Dockerfile:19]: Done.

File: .github/workflows/docker.yml (hunk 1 above)
Comment (pechouc) [reply .github/workflows/docker.yml:17]: This is determined by the command in the `Dockerfile`:

```
CMD ["gunicorn", "-b 0.0.0.0:80", "src.app:server"]
```

With which the app runs on port 80.

File: requirements.txt (hunk 3 above)
Comment (frgfm) [reply requirements.txt:1]: pandas is still there, you can remove it :sweat_smile:

**Review by frgfm — CHANGES_REQUESTED** [review]
Thanks! Only need to remove pandas and we're good to go :)

**Review by frgfm — COMMENTED** [review]
All good, but could you:
- add a screenshot in the PR description to document visual changes
- remove the empty space on the right & bottom of the page? It looks like we have a lot of empty space, which we can add back later on when we'll need it (and on desktop, that will look better)

Thanks!

**Comment by pechouc** [comment]
- Changes done for the spaces in the layout
- Re-deployed the app on Heroku
- Added pictures for documentation in the first message above

Thanks for the comments!

**Review by frgfm — APPROVED** [review]
Looks perfect :)

**Pull request merged**